libs/algo/odc/algo/_percentile.py

```diff
@@ -65,11 +67,11 @@ def xr_percentile(
         for percentile in percentiles:
             name = f"{band}_pc_{int(100 * percentile)}"
             yy = da.map_blocks(
-                lambda x: np_percentile(x, percentile, nodata, high=high), 
+                partial(np_percentile, percentile=percentile, nodata=nodata, high=high), 
                 xx_data, 
                 drop_axis=0, 
                 meta=np.array([], dtype=xx.dtype),
-                name=name,
+                name=randomize(name),
```

Comment:
`randomize` is kinda lazy way, we should not be using it anymore, I know it's all over the place, instead it's best to use `dask.base.tokenize` and give correct set of inputs.

example usage:
https://github.com/opendatacube/odc-tools/blob/29ea012d4bc431ad2bc02b558d3ee19619b74069/libs/algo/odc/algo/_memsink.py#L64

Reply:
in fact you can share token across sub-tasks, since they have unique names, so maybe something like

```python
tk = tokenize(xx.data, percentiles, nodata, high)
for percentile in percentiles:
      name = f"{band}_pc_{int(100 * percentile)}-{tk}"
```